src/lib/modules/blockchain_connector/index.js

```diff
@@ -411,12 +413,23 @@ class BlockchainConnector {
       'get',
       '/embark-api/blockchain/transactions/:hash',
       (req, res) => {
-        self.getTransaction(req.params.hash, (err, transaction) => {
-          if (err) {
-            self.logger.error(err);
-          }
-          res.send(transaction);
-        });
+        const isRawTxHash = req.query.isRawTxHash && req.query.isRawTxHash === 'true';
+
+        if (isRawTxHash) {
+          self.getTransactionByRawTransactionHash(req.params.hash, (err, transaction) => {
+            if (err) {
+              self.logger.error(err);
+            }
+            res.send(transaction);
```

Comment:
If there was an error, this will send `null`. Should probably send the error if there's one.

Reply:
This is a fair point. Just to clarify what's going on here though, we've never sent an actual error to the client in these APIs because the client simply assumes a certain data record inside redux. If it doesn't exist, it'll render and error. 

So with that in mind, I guess we'll have to choose between:

1. Staying consistent with the rest (or changing them all) and don't send a error because the client handles error cases already
2. Actually send an error and dedicated failure code so that redux' error handler is triggered in the client.

Reply:
Ah ok, I understand. If we end up sending right now an error code, example, 404, does the app handle it well? By that I mean, it doesn't "crash"?

Reply:
Yes, if we provide an error message it should be fine.